torch/distributed/checkpoint/planner.py

```diff
@@ -426,3 +427,31 @@ def commit_tensor(self, read_item: ReadItem, tensor: torch.Tensor) -> None:
         The contents of tensor will follow its device synchronization model.
         """
         pass
+
+
+class Checkpointable:
```

Comment:
I wonder if we should make this public in the first place, or we want to keep it private first and public it later? @LucasLLC

Reply:
I'm totally not against a private first approach, which could de-risk this impl.

Reply:
Makes sense to me, made private

Reply:
Agreed, made private